### PR TITLE
Add suport for GHC nightlies via GHCup

### PR DESCRIPTION
**Pull request by wismill**
Available since GHCup 0.1.19.3.

Related issues: #93, #137

### CODE REVIEW

**Comment by wismill** [comment]
Currently failing. Apart from the Cabal version (fix prepared), it fails on the expected GHC version check. We have no way to know in advance this version!

Should we check only the `A.B` in `A.B.C`?

**Comment by wismill** [comment]
I found a way to check the exact version of the latest GHC nightly, but it seems convoluted.

Also, why `steps.setup.outputs.ghcup-release-channel` is not set when `matrix.ghcup_release_channel` is?

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -160,8 +167,18 @@ jobs:
           GHCVER="$(ghc --numeric-version)"
           echo "CABALVER=${CABALVER}" >> "${GITHUB_ENV}"
           echo "GHCVER=${GHCVER}"     >> "${GITHUB_ENV}"
+          if [[ "${{ steps.setup.outputs.ghc-version }}" == "latest-nightly" ]]
+          then
+            # TODO: Shouldn't we use steps.setup.outputs.ghcup-release-channel?
+            #       Why is empty?
```
Comment (andreasabel) [inline .github/workflows/workflow.yml:173]: `ghcup-release-channel` is not an output of the setup action, you can see inputs and outputs of this action here: https://github.com/haskell/actions/blob/fc675398fab930927d01e9c00eee89066aa2ca12/setup/action.yml

Comment (wismill) [reply .github/workflows/workflow.yml]: I see. So the current use of `matrix.ghcup_release_channel` is OK I guess?

**Review by andreasabel — APPROVED** [review]
Thanks, apologies for the delay!

**Comment by andreasabel** [comment]
Released as `v2.4.6`.

**Pull request merged**